cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1341 @@
+	confData.ListID = ""
+	confData.Key = ""
+	confData.Token = ""
+}
+
+// CanUpdateTrello checks if all the data necessary for updating trello is available
+func CanUpdateTrello() bool {
+	if AreAPIKeysSet() && IsTrelloBoardDataSet() {
+		return true
+	}
+	return false
+}
+
+// AreAPIKeysSet checks if api keys and tokens are set
+func AreAPIKeysSet() bool {
+	if (apiKey != "" && apiToken != "") || (configData.Key != "" && configData.Token != "") {
+		return true
+	}
+	return false
+}
+
+// IsTrelloBoardDataSet checks if data required to update trello board is set
+func IsTrelloBoardDataSet() bool {
+	if (trelloBoardID != "" && trelloListID != "" && trelloChecklistID != "" && trelloCardID != "") ||
+		(configData.CardID != "" && configData.ChecklistID != "" && configData.BoardID != "" && configData.ListID != "") {
+		return true
+	}
+	return false
+}
+
+// getSha gets the sha of the latest commit
+func getSha(repoPath string) (ShaResponse, error) {
+	var resp ShaResponse
+	path := fmt.Sprintf(githubPath, repoPath)
+	if verbose {
+		log.Printf("Getting SHA of this path: %v\n", path)
+	}
+	return resp, SendGetReq(path, &resp)
+}
+
+// CheckExistingExchanges checks if the given exchange exists
+func CheckExistingExchanges(exchName string, confData *Config) bool {
+	for x := range confData.Exchanges {
+		if confData.Exchanges[x].Name == exchName {
+			return true
+		}
+	}
+	return false
+}
+
+// CheckMissingExchanges checks if any supported exchanges are missing api checker functionality
+func CheckMissingExchanges(confData *Config) []string {
+	var tempArray []string
+	for x := range confData.Exchanges {
+		tempArray = append(tempArray, confData.Exchanges[x].Name)
+	}
+	supportedExchs := exchange.Exchanges
+	for z := 0; z < len(supportedExchs); {
+		if common.StringDataContainsInsensitive(tempArray, supportedExchs[z]) {
+			supportedExchs = append(supportedExchs[:z], supportedExchs[z+1:]...)
+			continue
+		}
+		z++
+	}
+	return supportedExchs
+}
+
+// ReadFileData reads the file data from the given json file
+func ReadFileData(fileName string) (Config, error) {
+	var c Config
+	file, err := os.Open(fileName)
+	if err != nil {
+		return c, err
+	}
+	defer file.Close()
+	byteValue, err := ioutil.ReadAll(file)
+	if err != nil {
+		return c, err
+	}
+	err = json.Unmarshal(byteValue, &c)
+	if err != nil {
+		return c, err
+	}
+	return c, nil
+}
+
+// CheckUpdates checks updates.json for all the existing exchanges
+func CheckUpdates(fileName string, confData *Config) error {
+	var resp []string
+	errMap := make(map[string]error)
+	var wg sync.WaitGroup
+	var m sync.Mutex
+	allExchangeData := confData.Exchanges
+	for x := range allExchangeData {
+		wg.Add(1)
+		go func(e ExchangeInfo) {
+			switch e.CheckType {
+			case github:
+				m.Lock()
+				repoPath := e.Data.GitHubData.Repo
+				m.Unlock()
+				sha, err := getSha(repoPath)
+				m.Lock()
+				if err != nil {
+					errMap[e.Name] = err
+				}
+				if sha.ShaResp != e.Data.GitHubData.Sha {
```

Comment:
at this point there is a chance that the sha returned from getSha() will be empty or invalid as we just continue code execution above if error is not nil